crates/bevy_ecs/src/component.rs

```diff
@@ -168,7 +169,7 @@ pub struct ComponentDescriptor {
     layout: Layout,
     // SAFETY: this function must be safe to call with pointers pointing to items of the type
     // this descriptor describes.
-    drop: for<'a> unsafe fn(OwningPtr<'a>),
+    drop: Option<for<'a> unsafe fn(OwningPtr<'a>)>,
```

Comment:
Can you add the `// None if the underlying type doesn't need to be dropped` comment here too?